src/utils/work/getExtraDependencies.js

```diff
@@ -17,12 +17,18 @@ const getClusteringSettings = async (experimentId, body, dispatch, getState) =>
   return clusteringSettings;
 };
 
-// Check that the cell sets within the selected cellSetKey didn't change
+// Check that the cell sets within the selected selectedCellSet didn't change
 // e.g., if cell set was deleted we can't use cache
 const getSelectedCellSet = async (experimentId, body, dispatch, getState) => {
+  // If not downsampling, then there's no dependency on groupedTracks
+  if (!body.downsampleSettings) return '';
+
   await dispatch(loadCellSets(experimentId));
 
-  const children = getCellSetsHierarchyByKeys([body.cellSetKey])(getState());
+  const [{ children }] = getCellSetsHierarchyByKeys(
+    [body.downsampleSettings.selectedCellSet],
+  )(getState());
+
   const cellSetsKeys = children.map((cellSet) => cellSet.key);
 
   return cellSetsKeys;
```

Comment:
in the function definition of `getCellSets` the parameter `body` is not used.

Why this check: `if (!body.downsampleSettings) return '';` is not needed in that function?

Reply:
Because `getCellSets` is not related to downsample settings, it's just checking if any change happens in the cell sets. 

If it did, then the ETag changes

Reply:
For example, if you look at this file, you can see `getCellSets` conditions the `DownloadAnnotSeuratObject`. This means that if the cell sets changed, then the annotated seurat object won't be the same as before (because the cell sets changed). 

As a consequence, the ETag changes accordingly so that we don't return the cached result from before the cell sets changed

Reply:
got it, thanks for explaining!